n_serial.c

```diff
@@ -84,60 +55,32 @@ const char *serialNoteTransaction(char *json, char **jsonResponse)
         }
     }
 
-    // Allocate a buffer for input, noting that we always put the +1 in the alloc so we can be assured
-    // that it can be null-terminated.  This must be the case because json parsing requires a
-    // null-terminated string.
-    int jsonbufAllocLen = ALLOC_CHUNK;
+    // Allocate a buffer for input, noting that we always put the +1 in the
+    // alloc so we can be assured that it can be null-terminated. This must be
+    // the case because json parsing requires a null-terminated string.
+    size_t jsonbufAllocLen = ALLOC_CHUNK;

```

Comment:
The code from here to line 96 is nearly identical to the corresponding code in i2cNoteTransaction. The only differences are

1. serialRawReceive is used here whereas we use i2cRawReceive in i2cNoteTransaction.
2. i2cNoteTransaction does this:
```
if (overflow > ALLOC_CHUNK) {
    jsonbufAllocLen += overflow;
} else {
    jsonbufAllocLen += ALLOC_CHUNK;
}
```

Is there a reason for the difference in 2? Regarding 1, is this an opportunity to unify the logic into one function used by both i2cNoteTransaction and serialNoteTransaction?